woocommerce/woocommerce-ios
# Add IPP plugin status tags to ZenDesk

Closes: #7994 

### Description
With this PR, we'll be sending extra tags to ZenDesk that reflect the plugin status of `woocommerce-gateway-stripe` and `woocommerce-payments` plugins. There are 3 tags for each plugin:

- `_installed_and_activated`
- `_installed_and_not_activated`
- `_not_installed`

### Changes

In order to add these IPP statuses to ZenDesk tickets, we need to pass [these tags](https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/Tools/Zendesk/ZendeskManager.swift#L625) through the SupportSDK's `RequestUiConfiguration` class when we create a new request. As we need to do this before ZenDesk tabs loads, we'll be using a `StorageSitePlugin` results controller directly through the `ZendeskManager` class, in order to fetch plugin statuses and pass these as tags.


### Testing instructions
The easiest way is through the debugger:
1. Confirm that you have [WooCommerce Payments](https://wordpress.org/plugins/woocommerce-payments/), and/or [WooCommerce Stripe Gateway](https://wordpress.org/plugins/woocommerce-gateway-stripe/) installed and/or activated
2. Add a breakpoint to [requestConfig.tags](https://github.com/woocommerce/woocommerce-ios/blob/b59a59d2efafc94dd861d15561cd1cac593da4b2/WooCommerce/Classes/Tools/Zendesk/ZendeskManager.swift#L631) to confirm which tags are we sending to ZenDesk.
3. Go to Menu > Settings > Help and Support > Tap on "Contact Support"
4. In Xcode's debugger, see the tags: 
```
tags	[String]	7 values	
[0]	String	"iOS"	
[1]	String	"woo-mobile-sdk"	
[2]	String	"jetpack"	
[3]	String	"woo_mobile_stripe_not_installed"	
[4]	String	"woo_mobile_wcpay_not_installed"	
[5]	String	"wpcom"	
[6]	String	"eCommerce"	
```
It can also be tested directly in ZenDesk, which requires extra steps:
1. Confirm that you have [WooCommerce Payments](https://wordpress.org/plugins/woocommerce-payments/), and/or [WooCommerce Stripe Gateway](https://wordpress.org/plugins/woocommerce-gateway-stripe/) installed and/or activated
2. Go to Menu > Settings > Help and Support > Confirm that you're logged into the app with a non-a8c account > Tap on "Contact Support" and send a request (You may need to test this on a physical device, not the simulator)
3. Log into ZenDesk, find your ticket, and confirm that the tags have been added and reflect the plugin status. If you have no ZD account, feel free to ping me on Slack with the non-a8c-email used (or any HE :D), in order to find the ticket.

<img width="355" alt="Screenshot 2022-11-07 at 12 30 43" src="https://user-images.githubusercontent.com/3812076/200220826-63d6f806-2ab5-4cc4-8cf5-0a11e5b4b841.png">

---
